aten/src/ATen/native/TensorIterator.cpp

```diff
@@ -734,6 +746,8 @@ int TensorIterator::get_dim_to_split() const {
 void TensorIterator::build() {
   // set is_output and is_read_write flags on appropriate tensors
   mark_outputs();
+  // Check outputs' internal overlaps and input-output overlaps
```

Comment:
A more descriptive comment here would be nice. Something like 
```
// Check that the outputs have no internal overlap and do not share memory with inputs.
```

Reply:
Addressed